File: go/blackjack/blackjack.go
package blackjack

// ParseCard returns the integer value of a card following blackjack ruleset.
func ParseCard(card string) int {
	switch card {
	case "ace":
		return 11
	case "two":
		return 2
	case "three":
		return 3
	case "four":
		return 4
	case "five":
		return 5
	case "six":
		return 6
	case "seven":
		return 7
	case "eight":
		return 8
	case "nine":
		return 9
	case "ten", "jack", "queen", "king":
		return 10
	}
	return 0
}

func FirstTurn(card1, card2, dealerCard string) string {

	handScore := ParseCard(card1) + ParseCard(card2)
	isBlackJack := IsBlackjack(card1, card2)
	dealerScore := ParseCard(dealerCard)

	if handScore > 20 {
		return LargeHand(isBlackJack, dealerScore)
	}
	return SmallHand(handScore, dealerScore)
}

// IsBlackjack returns true if the player has a blackjack, false otherwise.
func IsBlackjack(card1, card2 string) bool {
	cardValue1, cardValue2 := ParseCard(card1), ParseCard(card2)
	return cardValue1+cardValue2 == 21
}

// LargeHand implements the decision tree for hand scores larger than 20 points.
func LargeHand(isBlackjack bool, dealerScore int) string {
	switch {
	case !isBlackjack:
		return "P"
	case dealerScore != 11 && dealerScore != 10:
		return "W"
	}
	return "S"
}

// SmallHand implements the decision tree for hand scores with less than 21 points.
func SmallHand(handScore, dealerScore int) string {
	switch {
	case handScore >= 17:
		return "S"
	case handScore <= 11, dealerScore >= 7:
		return "H"
	}
	return "S"
}

File: java/blackjack/src/main/java/Blackjack.java
public class Blackjack {
    private static final String STAND = "S";
    private static final String HIT = "H";
    private static final String SPLIT = "P";
    private static final String WIN = "W";

    public int parseCard(String card) {
        return switch (card) {
            case "ace" -> 11;
            case "jack", "queen", "king", "ten" -> 10;
            case "nine" -> 9;
            case "eight" -> 8;
            case "seven" -> 7;
            case "six" -> 6;
            case "five" -> 5;
            case "four" -> 4;
            case "three" -> 3;
            case "two" -> 2;
            default -> 0;
        };
    }

    public boolean isBlackjack(String card1, String card2) {
        return parseCard(card1) + parseCard(card2) == 21;
    }

    public String largeHand(boolean isBlackjack, int dealerScore) {
        if (!isBlackjack) {
            return SPLIT;
        }

        if (dealerScore != 10 && dealerScore != 11) {
            return WIN;
        }

        return STAND;
    }

    public String smallHand(int handScore, int dealerScore) {
        if (handScore >= 17) {
            return STAND;
        }

        if (handScore <= 11 || dealerScore >= 7) {
            return HIT;
        }

        return STAND;
    }

    // FirstTurn returns the semi-optimal decision for the first turn, given the cards of the player and the dealer.
    // This function is already implemented and does not need to be edited. It pulls the other functions together in a
    // complete decision tree for the first turn.
    public String firstTurn(String card1, String card2, String dealerCard) {
        int handScore = parseCard(card1) + parseCard(card2);
        int dealerScore = parseCard(dealerCard);

        if (20 < handScore) {
            return largeHand(isBlackjack(card1, card2), dealerScore);
        } else {
            return smallHand(handScore, dealerScore);
        }
    }
}
